### PR TITLE
Navview fix issue with compact pane length not being behaving correctly

### LINKED ISSUE #2277: Setting CompactPaneLength before rendering in XAML does not update pane correctly
**Describe the bug**

**Steps to reproduce the bug**

1. Create NavView in XAML, set CompactPaneLength to a non standard value, e.g. 80
2. Open page with that NavView

**Expected behavior**

TogglePaneButton and BackButton get rendered correctly, that is the button icons are centered and the buttons take up the same space as the NavigationViewItems icons.
**Screenshots**

image

**Version Info**

NuGet package version: 

Latest commit on master branch.

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (19603)              | Yes   |
| November 2019 Update (18363)  |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes  |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Fix issue when we set the CompactPaneLength before the page has finished loading.
## Motivation and Context

Closes #2277 
## How Has This Been Tested?

Update existing interaction test.

### CODE REVIEW

File: dev/NavigationView/NavigationView_InteractionTests/NavigationViewTests.cs [hunk 1]
```diff
@@ -4366,17 +4366,22 @@ public void VerifyIconsRespectCompactPaneLength()
                 var currentStatus = new CheckBox(FindElement.ByName("MenuItemsCorrectOffset"));
 
                 checkMenuItemsButton.Click();
+                Log.Comment("Verifying compact pane length set before loading working");
                 Wait.ForIdle();
+                compactpaneCheckbox = new ComboBox(FindElement.ByName("CompactPaneLenghtComboBox"));
```
Comment (karenbtlai) [inline dev/NavigationView/NavigationView_InteractionTests/NavigationViewTests.cs:4371]: >CompactPaneLenghtComboBox [](start = 71, length = 25)

nit: I realize this is a previous typo, but can be fixed?

Comment (chingucoding) [reply dev/NavigationView/NavigationView_InteractionTests/NavigationViewTests.cs]: Good idea, fixed.

File: dev/NavigationView/NavigationView.cpp [hunk 2]
```diff
@@ -3496,6 +3496,7 @@ void NavigationView::OnLoaded(winrt::IInspectable const& sender, winrt::RoutedEv
         m_titleBarMetricsChangedRevoker = coreTitleBar.LayoutMetricsChanged(winrt::auto_revoke, { this, &NavigationView::OnTitleBarMetricsChanged });
         m_titleBarIsVisibleChangedRevoker = coreTitleBar.IsVisibleChanged(winrt::auto_revoke, { this, &NavigationView::OnTitleBarIsVisibleChanged });
     }
+    UpdatePaneButtonsWidths();
```
Comment (ojhad) [inline dev/NavigationView/NavigationView.cpp:3499]: nit: maybe adding a comment here to explain why this needs to be called in OnLoaded will be helpful

Comment (chingucoding) [reply dev/NavigationView/NavigationView.cpp:3500]: Good idea, done.

**Pull request merged**